doc/specs/stdlib_linalg.md

```diff
@@ -101,21 +101,29 @@ end program demo_diag5
 
 Experimental
 
+### Class
+
+Pure function.
+
 ### Description
 
-Construct the identity matrix
+Construct the identity matrix.
 
 ### Syntax
 
-`I = [[stdlib_linalg(module):eye(function)]](n)`
+`I = [[stdlib_linalg(module):eye(function)]](dim1 [, dim2])`
 
 ### Arguments
 
-`n`: Shall be a scalar of default type `integer`. 
+`dim1`: Shall be a scalar of default type `integer`.
+This is an `intent(in)` argument. 
+
+`dim2`: Shall be a scalar of default type `integer`.
+This is an `intent(in)` and `optional` argument. 
 
 ### Return value
 
-Returns the identity matrix, i.e. a square matrix with ones on the main diagonal and zeros elsewhere. The return value is of type `integer(int8)`.
+Returns the identity matrix, i.e. a square matrix with ones on the main diagonal and zeros elsewhere. The return value is of type `integer`.
```

Comment:
The matrix may not be square since dim1 may differ from dim2

Reply:
Why the results has to be type `integer`? what about `real`? what about other kinds? I remember python has eye function for different type.

Reply:
The original reasons for having only an integer version can be found here https://github.com/fortran-lang/stdlib/issues/169#issuecomment-611074126. Other kinds are achieved by implicit or explicit type promotion.

Some other possibilities are explored in the thread @zoziha created at Discourse: https://fortran-lang.discourse.group/t/fortran-function-return-value-polymorphism/1350

The best solution would be to submit a proposal to J3 to allow specifying an optional return kind for function results, see https://github.com/j3-fortran/fortran_proposals/issues/91.

Reply:
The reasoning in the comment is only valid to the `eye` function, isn't it?

Reply:
> Why the results has to be type `integer`? what about `real`? what about other kinds? I remember python has eye function for different type.

Such an `eye` function can be converted to the `real/complex` type (see https://fortran-lang.discourse.group/t/fortran-function-return-value-polymorphism/1350/11?u=zoziha), but this is indeed an experimental solution. I hope that such a proposal (see https://github.com/j3-fortran/fortran_proposals/issues/91) can be passed.